polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -281,6 +282,14 @@ $loading-panel-height: 53px;
     var(--p-duration-100);
 }
 
+.TableHeadingUnderline {
+  border-bottom: var(--p-border-width-2) dotted var(--p-border-subdued);
+}
+
+.TableHeadingTooltipUnderlinePlaceholder {
+  border-bottom: var(--p-border-width-2) dotted transparent;
+}
```

Comment:
This placeholder underline maintains the underline space as shown in @kyledurand's comment [here](https://github.com/Shopify/polaris/pull/8277#pullrequestreview-1287399308).

This is only used on the `sortMarkup` when we have `sortToggleLabels`, we are in the `selectMode` (the row has been clicked) and we have `tooltipContent`. We check for `tooltipContent` since when we are not in the select mode, we render the tooltip with an underline.